File: 1704-special-positions-in-a-binary-matrix/special-positions-in-a-binary-matrix.py
# class Solution:
#     def numSpecial(self, mat: List[List[int]]) -> int:
#         def get_column_sum(col_idx):
#             return sum(row[col_idx] for row in mat)

#         special = 0
#         for row in mat:
#             if sum(row) == 1:
#                 col_idx = row.index(1)
#                 special += get_column_sum(col_idx) == 1

#         return special
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        ans = 0
        m = len(mat)
        n = len(mat[0])
        
        for row in range(m):
            for col in range(n):
                if mat[row][col] == 0:
                    continue
                    
                good = True
                for r in range(m):
                    if r != row and mat[r][col] == 1:
                        good = False
                        break
                
                for c in range(n):
                    if c != col and mat[row][c] == 1:
                        good = False
                        break
                
                if good:
                    ans += 1
        
        return ans

File: 1704-special-positions-in-a-binary-matrix/special-positions-in-a-binary-matrix.cpp
// #include <bits/stdc++.h>
// using namespace std;

// class Solution {
// public:
//     int numSpecial(vector<vector<int>>& mat) {
//         int res=0;
//         int m=mat.size();
//         int n=mat[0].size();

//         for(int row=0;row<m;row++){
//             for(int col=0;col<n;col++){
//                 if(mat[row][col]==0) continue;
//                 bool special=true;
//                 //check col
//                 for(int r=0;r<m;r++){
//                     if(r!=row && mat[r][col]==1){
//                         special=false;
//                         break;
//                     }
//                 }
//                 //check row
//                 for(int c=0;c<n;c++){
//                     if(c!=col && mat[row][c]==1){
//                         special=false;
//                         break;
//                     }
//                 }
//                 if(special==true){
//                     res++;
//                 }
//             }
//         }
//         return res;
//     }
// }; 
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        // int res=0;
        int m=mat.size();
        int n=mat[0].size();

        vector<int> rowcount(m,0); //how many 1s in a row
        vector<int> colcount(n,0); //how many 1s in a col

        for(int row=0;row<m;row++){
            for(int col=0;col<n;col++){
                if(mat[row][col]==1){
                    rowcount[row]++;
                    colcount[col]++;
                }
            }    
        }
        int res=0;
        for(int row=0;row<m;row++){
            for(int col=0;col<n;col++){
                if(mat[row][col]==0) continue;
                if(rowcount[row]==1 && colcount[col]==1){
                    res++;
                }
            }
        }
        return res;
    }
};
